go/mysql/conn.go

```diff
@@ -1103,15 +1118,14 @@ func (c *Conn) handleNextCommand(handler Handler) error {
 		if err := c.writePrepare(fld, c.PrepareData[c.StatementID]); err != nil {
 			return err
 		}
-
 	case ComStmtExecute:
 		// flush is called at the end of this block.
 		// To simplify error handling, we do not
 		// encapsulate it with a defer'd func()
 		c.startWriterBuffering()
 
 		queryStart := time.Now()
-		stmtID, _, err := c.parseComStmtExecute(c.PrepareData, data)
+		stmtID, cursorType, err := c.parseComStmtExecute(c.PrepareData, data)
```

Comment:
I think that makes sense for `ComStmtExecute`, since that would open another cursor, potentially causing problems, especially since we're only going to support one cursor at a time. I don't think `ComStmtPrepare` would be affected by any open cursors. `ComStmtQuery` might cause problems if the query affects the table the cursor is reading from, like inserting or updating rows.

Reply:
Yeah, I think there are situations where it would be fine to let Prepares run, but my concerns is with adding concurrency on the session when there wasn't any previously. For example, I'm pretty sure the database catalog has a transactional view of the schemas of the tables being queried, and that's referenced in the analyzer. It seems most expedient to just disallow it for now, assuming existing clients don't rely on this.